pkg/controller/handlers/retention/retention.go

```diff
@@ -0,0 +1,72 @@
+package retention
+
+import (
+	"time"
+
+	"github.com/obot-platform/nah/pkg/router"
+	"github.com/obot-platform/obot/logger"
+	v1 "github.com/obot-platform/obot/pkg/storage/apis/obot.obot.ai/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var log = logger.Package()
+
+type Manager struct {
+	c      kclient.Client
+	policy time.Duration
+}
+
+func NewRetentionManager(c kclient.Client, policy time.Duration) *Manager {
+	if policy == 0 {
+		log.Infof("retention policy: disabled")
+	} else {
+		log.Infof("retention policy: %s", policy)
+	}
+
+	return &Manager{
+		c:      c,
+		policy: policy,
+	}
+}
+
+func (m *Manager) Run(req router.Request, resp router.Response) error {
+	if m.policy == 0 {
+		return nil
+	}
+
+	thread := req.Object.(*v1.Thread)
+	if thread.Spec.SystemTask {
+		return nil
+	}
+
+	if thread.Spec.Project {
+		// If this thread is a project, there is a chance it is a featured Obot.
+		// We do not want to clean up featured Obots. Check the thread shares to see if it is one.
+		shares := &v1.ThreadShareList{}
+		if err := m.c.List(req.Ctx, shares, kclient.InNamespace(thread.Namespace), &kclient.ListOptions{
+			FieldSelector: fields.SelectorFromSet(map[string]string{
+				"spec.featured":          "true",
```

Comment:
Remove this line, and check each item to see if it is featured.

Reply:
Curious what the reason for this is? Is it more performant to just cycle through them in a loop instead of using the field selector?

Reply:
(fixed btw)

Reply:
The cached client doesn't allow you to list with multiple field selectors. So, you have list with one and adjust.